classes/phing/tasks/system/condition/DatabaseCondition.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * Returns false when the connection to a database fails, and true otherwise.
+ *
+ * @link    https://www.php.net/manual/en/pdo.drivers.php
+ * @link    https://www.php.net/manual/en/ref.pdo-mysql.connection.php
+ * @link    https://www.php.net/manual/en/ref.pdo-pgsql.connection.php
+ *
+ * @author  Jawira Portugal <dev@tugal.be>
+ * @package phing.tasks.system.condition
+ */
+class DatabaseCondition extends ProjectComponent implements Condition
+{
+    protected $dsn      = null;
+    protected $username = null;
+    protected $password = null;
+
+    public function setDsn(string $dsn): void
+    {
+        $this->dsn = $dsn;
+    }
+
+    public function setUsername(string $username): void
+    {
+        $this->username = $username;
+    }
+
+    public function setPassword(string $password): void
+    {
+        $this->password = $password;
+    }
+
+    public function evaluate(): bool
```

Comment:
Please move this method to the `PDOSQLExecTask` and let it `implements Condition` like the `PhingVersion` task/condition does.

Reply:
great suggestion, I'm working on it!

Reply:
Done, can you check it please? As usual I'm open to any question/suggestion.